lib/player.js

```diff
@@ -2701,7 +2721,17 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
         this.manifest_,
         this.config_.streaming,
         startTime,
-        () => this.onSeek_());
+        () => this.onSeek_(),
+        (event) => {
+          if (event.type === 'stalldetected') {
```

Comment:
My only issue with the code is that using a string literal internally defeats the purpose of the enum, which is meant to help us avoid typos in string literals.

But I would guess you did this to avoid a circular dependency between Player and the infernal components, and I'm not sure what the best solution is.  I'm reviewing this on my phone and I haven't dug into the code yet.

One idea that occurs to me now is that we could move the enum to another file that Player, GapJumpingController, and StallDetector could all require.  Since EventName isn't exported, that won't break any public API surface.

Reply:
> One idea that occurs to me now is that we could move the enum to another file that Player, GapJumpingController, and StallDetector could all require.

Sounds good, I think a good place to move the enum is in the utils/ folder, specifically in the fake_event.js file - let me know what you think!

Also maybe I'm not understanding circular dependencies correctly, but why is StreamingEngine allowed to reference the enum namespace without requiring it (https://shaka-player-demo.appspot.com/docs/api/lib_media_streaming_engine.js.html#line1697), but if I try to reference it in GapJumpingController or StallDetector the build complains?

Reply:
It shouldn't be allowed, but we didn't goog.require it.  If we did, it would fail to build, because it would be an explicit circular dependency.  I don't know if this was done on purpose or just overlooked.  Probably the latter.

I think it's best to avoid undeclared circular deps, too, even if the compiler would let us get away with it.